copier/main.py

```diff
@@ -326,6 +330,10 @@ def answers(self) -> AnswersMap:
             if var_name in result.init:
                 # Do not ask again
                 continue
+
+            with suppress(KeyError):
+                details['default'] = self.default_overrides[var_name]
```

Comment:
Sorry, I just realized that this logic should be in the right spot: 

https://github.com/copier-org/copier/blob/fefc5921f989bb602fecbff8a48b89bfa2ab7266/copier/user_data.py#L70

That class is in charge of choosing the correct answer from different sources, you're just introducing a new source.

Reply:
Just occurred to me that merging default_overrides and template_defaults near here: https://github.com/copier-org/copier/blob/overriden-defaults/copier/main.py#L323 and simply passing `defaults` feels best.

It feels odd having multiple sources in AnswersMap. At least, from the POV of the AnswersMap API having both `defaults` and `default_overrides` feels odd. Do you get default_override_overrides as a third tier at some point?

Reply:
LMK how you feel about that. IMO reconciliation of these overrides feels like glue code well suited main / Worker. AnswersMap is a pretty straightforward data class at the moment.

If you believe this is an `AnswersMap` thing, it would make sense to generalize / expand `defaults` into some multi-tiered thing such as an ordered list of `DefaultSources` rather than a single default. 

E.g. tidy up this implementation by doing it here (as mentioned) https://github.com/copier-org/copier/blob/overriden-defaults/copier/main.py#L323

If AnswersMap gets multiple default_sources in the future, then main.py no longer has to merge template defaults and default_overrides and could just pass something like:

```python
AnswersMap(
    default=self.template.defaults, 
    additional_defaults=[
        DefaultSource('user_configuration', self.user_defaults),
        DefaultSource('default_overrides', self.default_overrides),
    ]
)
```

Reply:
Of course, as I try it I see it won't work because the `Question` is created independently using different `question_data`.

Reply:
Well, I still feel `AnswersMap` is the place where it should go, as its purpose for existence is to prioirze and merge different answer sources.

Regarding the change, I'd just change `defaults` for `template_defaults`, and add a new `user_defaults` that takes priority over `template_defaults`.

Reply:
Alright, I see it. I'll have something.

Reply:
https://github.com/copier-org/copier/pull/477/commits/8d3e90523b34572bbaa0e1874433511e2ba342ae

This look about right? Not entirely clear where we want the `self.user_defaults` in the `AnswersMap:combined`. Existing tests pass but `AnswersMap` might benefit from some more finagling now.